### ISSUE TITLE
[FEAT.] : Cleanup duplicate songs

### ISSUE THREAD

**Issue by f1amy** [Zagrios/bs-manager#593]
## Problem

Playing BS with custom songs for a long time created a problem of having downloaded multiple copies of the same maps, which probably happened when using different installation methods (BSManager and ModAssistant, for example).
Some of the copies could be outdated map versions, but that's a whole another issue.

In the game this results in showing multiple instances of the same song when searching, with only one of them showing correct leaderboards (BeatLeader, might be their bug tho).

## Solution

The way I see it, there could be a button in BSManager to cleanup/normalize downloaded songs in `BSManager/SharedContent/SharedMaps/CustomLevels`.

It could read each song's short ID, name, artists from the metadata and remove redundant copies.

## Additional context

Don't know if playing this map will result in score upload since it does not show map ranked status
![image](https://github.com/user-attachments/assets/64bb74d9-e22b-46d0-b803-4b3efe60a3a8)
The last one that shows all leaderboard info
![image](https://github.com/user-attachments/assets/0e1f054d-8e42-4403-b09d-d669c5f47bb8)

**Comment by f1amy** [comment]
I play on Windows btw, not sure if OS is relevant here

**Comment by silentrald** [comment]
Are the folder names of the maps/songs patterned like this:

```
id-song
id-song (2)
id-song (3)
```

Or are they very different to one another?

---

Also the labeler just tagged this as linux because it found "arch" in the word "searching" probably.

**Comment by Zagrios** [comment]
@silentrald I don't think that matters. One map can be named cleanly and still be duplicated in a folder named AAAAAA 😅\
We can just scan for duplicate hashes\
I'll take this one; I've already done some work on it but never committed it

**Comment by f1amy** [comment]
> Are the folder names of duplicate maps/songs patterned like this:
> ```
> id-song
> id-song (2)
> id-song (3)
> ```
> Or are they very different to one another?

I'll add for the context these were the patterns
1. same name with `(1)` at the end
2. one is "hash (song name - mappers)", second "hash (mappers - song name)"
3. same as 2. but song name or mappers are truncated
4. one is as from 2. and truncated, second is in some other format, like "hash-full song name from beat saver", like "[Chroma+] Bad Side - Jake Daniels"

there might be other patterns
